### PR TITLE
Remove grouping from FungibleAsset (Issue and Move)

### PR DESCRIPTION
**Pull request by kchalkias**
This is part of the [finance CorDapp improvements](https://discourse.corda.net/t/various-things-to-improve-in-the-finance-cordapp/1386)
IssueCommand is also totally removed as no nonce is required anymore and all Issues can be `TypeOnlyCommandData`

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt [hunk 1]
```diff
@@ -32,23 +32,8 @@ interface FungibleAsset<T : Any> : OwnableState {
...
-        interface Issue : IssueCommand, Commands
-
-        /**
-         * A command stating that money has been withdrawn from the shared ledger and is now accounted for
-         * in some other way.
-         */
-        interface Exit<T : Any> : Commands {
-            val amount: Amount<Issued<T>>
-        }
+    interface ExitCommand<T : Any>: CommandData {
```
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: Why can't this go as well? No API declared on FungibleAsset consumes it

File: core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt [hunk 2]
```diff
@@ -32,23 +32,8 @@ interface FungibleAsset<T : Any> : OwnableState {
 
     fun move(newAmount: Amount<Issued<T>>, newOwner: AbstractParty): FungibleAsset<T>
```
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: If the move function stays it needs documenting.

File: core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt (hunk 1 above)
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: If it stays then please add document comments what this is for and why it has the amount for people less familiar with Corda.

File: core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt (hunk 2 above)
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: I was meant to touch this one in this PR as the initial purpose was to get rid of the grouping interface and remove Issue (due to nonce not required anymore), but I will take the opportunity.

File: core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt (hunk 1 above)
Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: Well, it's used in the same way by all `FungibleAsset` States in finance and unlike Move and Issue I decided to leave it here. We also have a `MoveCommand` in `Structures`, I could send it there, but it's probably dedicated to FungibleAssets as we've discussed already.

Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: I don't really see any value in keeping it. By convention my verify method should respect it, but it isn't enforced. I could also model exit as a flag on a terminal output state, or setting owner == issuer, or a whole host of other techniques.

Comment (kchalkias) [inline core/src/main/kotlin/net/corda/core/contracts/FungibleAsset.kt]: Ok, convinced. Anyway, it's just a data class with one value. Is there a real world scenario where a Fungible Asset shouldn't support Exit? I mean should we enforce it or not?

**Comment by kchalkias** [comment]
back to you @mnesbit, didn't touch `Amount<Issued<T>>` yet, will do in an upcoming PR.

**Pull request merged**